File: Leetcode/longest_common_prefix.py
class Solution:
    def longestCommonPrefix(self, strs) -> str:
        length = len(strs)
        if length == 1:
            return strs[0]
        if length == 0:
            return ""
        
        longest_prefix = strs[0]

        i = 1
        while i < len(strs):
            j = 0
            temp = ""
            while j < len(strs[i]) and j < len(longest_prefix):
                if strs[i][j] == longest_prefix[j]:
                    temp += longest_prefix[j]
                elif temp == "":
                    return ""
                else:
                    break
                j += 1
            longest_prefix = temp if len(temp) < len(longest_prefix) else longest_prefix
            i += 1
        return longest_prefix

File: LeetCode/LongestCommonPrefix.cs
public class Solution
{
    public string LongestCommonPrefix(string[] strs)
    {
        int len = strs.Length;
        if (len == 0)
        {
            return "";
        }
        if (len == 1)
        {
            return strs[0];
        }
        string longestCommonPrefix = strs[0];
        for (int i = 1; i < len; ++i)
        {
            string temp = "";
            int j = 0;
            while (j < strs[i].Length && j < longestCommonPrefix.Length)
            {
                if (strs[i][j] == longestCommonPrefix[j])
                {
                    temp += longestCommonPrefix[j];
                }
                else if (temp == "")
                {
                    return "";
                }
                else
                {
                    break;
                }
                ++j;
            }
            if (temp.Length < longestCommonPrefix.Length)
            {
                longestCommonPrefix = temp;
            }
        }
        return longestCommonPrefix;
    }
}
